andela/ah-codeblooded-frontend
# #160577586 Enable users to filter articles

**What does this PR do?**

It enables users to filter articles using specific parameters.

**Description of Task to be completed?**

Users may want to get desired articles quickly. This PR allows them to filter articles using parameters i.e tags and authors' usernames.

**How should this be manually tested?**

- Open this PR's review app
- Click on the search icon in the Navbar
- Click on filter icon
- Fill details, i.e tag or author or both
- The articles should appear as desired i.e (you will get articles by your desired author or tags)

**Any background context you want to provide?**

This feature is heavily integrated with the search feature. By this I mean that one can filter using a parameter and then search anything on the results that will be provided.

**What are the relevant pivotal tracker stories?**

https://www.pivotaltracker.com/story/show/160577586

**Screenshots (if appropriate)**

This is what you see when you navigate to the search page
<img width="1440" alt="screen shot 2018-11-29 at 11 29 29" src="https://user-images.githubusercontent.com/26184534/49208926-3270d980-f3ca-11e8-9d40-e586f3b414b5.png">

Input the filter parameters. There is also autocomplete as a user types
<img width="1440" alt="screen shot 2018-11-29 at 11 29 55" src="https://user-images.githubusercontent.com/26184534/49209044-7237c100-f3ca-11e8-995f-23c5f76b13d4.png">

Click apply and the articles should appear on your right
<img width="1440" alt="screen shot 2018-11-29 at 11 30 09" src="https://user-images.githubusercontent.com/26184534/49209113-a57a5000-f3ca-11e8-891b-eea3a5a7a082.png">

**Questions:**

N/A
